src/indexed_crate.rs

```diff
@@ -765,6 +773,39 @@ fn build_export_name_index(index: &HashMap<Id, Item>) -> HashMap<&str, &Item> {
     .collect()
 }
 
+fn build_variant_name_index(index: &HashMap<Id, Item>) -> HashMap<(Id, &str), (&Item, usize)> {
+    #[cfg(feature = "rayon")]
+    let iter = index.par_iter().map(|(_, value)| value);
+    #[cfg(not(feature = "rayon"))]
+    let iter = index.values();
+
+    iter.filter_map(|item| match &item.inner {
+        rustdoc_types::ItemEnum::Enum(e) => Some((item.id, e)),
+        _ => None,
+    })
+    .flat_map(|(id, enum_item)| {
+        #[cfg(feature = "rayon")]
+        let base_iter = enum_item.variants.par_iter();
+        #[cfg(not(feature = "rayon"))]
+        let base_iter = enum_item.variants.iter();
+
+        base_iter.copied().enumerate().map(move |(i, var_id)| {
+            let var = index.get(&var_id).expect("Variant id should be correct.");
```

Comment:
rustdoc JSON occasionally has bugs that result in dangling IDs, and some of those issues are still open. We _shouldn't_ hit this panic, but we _might_.

Slight preference for using `filter_map()` and omitting invalid items from the index instead of exploding, just so we don't cause ourselves extra support burden if such a bug affects our users.

Reply:
I have made the change, but I'm not convinced that it's a good idea. If dangling ids do exist, then it's likely that they will cause a false positive in `enum_variant_missing`. Then, instead of being able to point to the backtrace and easily identify the cause as dangling ids in rustdoc, there will be no clear signal that determines the cause.